content/blog/2023/2023-10-10-intro-to-mockk.md

```diff
@@ -0,0 +1,169 @@
+```kotlin
+@SpyK
+val calculator = Calculator()
+```
+
+**@UnmockK**: This annotation is used to unmock a property or object that was previously declared as a mock using `@MockK` or similar annotations. This is useful when we need to revert a mock back to its original behavior. 
+
+```kotlin
+@UnmockK
+lateinit var unmockedService: SomeService
+```
+
+## Mockk Keywords
+
+When using the MockK library for mocking and verifying interactions in Kotlin tests, there are several essential keywords and functions we should be familiar with. Here are some of the most commonly used keywords and functions in MockK:
+
+| Keyword | Description |
+|--------|---------------------------|
+|mockk()        | Usually creates a mock object of a given class or interface |
+|every{}        | Defines a behavior for mock object methods. We can specify what a method should return when invoked|
+|justRun{}    | Defines a behavior for a method without returning a value. Useful for methods with a Unit return type|
+|slot{}         | Captures arguments passed to a mocked method. We can use this to verify arguments later|
+|verify{}       | Verify that a method was called with specific arguments and a certain number of times|
+|atLeast(),atMost(),exactly()         | These keywords are used with `verify` to specify the number of times a method should be called.|
+|verifyOrder{}  | Verify the order in which methods were called on mock objects|
+|confirmVerified()| Ensures that all interactions with the mock have been verified. This is useful to prevent false positives in our tests|
+|clearMocks()   | Used to reset the verification state of one or more mock objects|
+|unmockkAll()   | Used to unmock all the mock objects created with the `mockk()` annotation|
+
+## Combining Mockk With Other Testing Libraries
+Combining JUnit and MockK is a popular approach for testing Kotlin code. JUnit is a widely used testing framework for Java and Kotlin, while MockK is a mocking library specifically designed for Kotlin. Together, they allow us to write comprehensive unit tests for our Kotlin code with mock objects. Here's how we can use JUnit and MockK for testing Kotlin code:
+```kotlin
+class CalculatorTest {
+    private lateinit var calculator: Calculator
+
+    private lateinit var mathService: MathService
+
+    @BeforeEach
+    fun setUp() {
+        mathService = mockk()
+        calculator = Calculator(mathService)
+    }
+
+    @Test
+    fun testAddTwoNumbers() {
+        every { mathService.add(2, 3) } returns 5
+        val result = calculator.addTwoNumbers(2, 3)
+        //using JUnit assertions 
+        assert(result == 5)
+    }
+}
+```
+In this test class, we've created a mock `MathService` using MockK and defined its behavior using `every`. We specify that when mathService.add(2, 3) is called, it should `return` 5. Then, we test the `addTwoNumbers` method of the `Calculator` class, and finally, we `assert` that the result is as expected.
```

Comment:
In this paragraph, you only explain the MockK features, and not the JUnit integration. It should explain how MockK works with JUnit, not re-iterate the MockK features that were already explained above.